src/core/execute.c

```diff
@@ -948,8 +949,13 @@ static int setup_pam(
                  * and this will make PR_SET_PDEATHSIG work in most cases.
                  * If this fails, ignore the error - but expect sd-pam threads
                  * to fail to exit normally */
+
+                if (setgroups(0, NULL) < 0)
+                        log_warning_errno(errno, "Failed to setgroups() in sd-pam: %m");
+                if (setresgid(gid, gid, gid) < 0)
+                        log_warning_errno(errno, "Failed to setresgid() in sd-pam: %m");
```

Comment:
`setresgid` doesn't respect the `SupplementaryGroups` ([`enforce_groups`](https://github.com/systemd/systemd/commit/096424d1230e0a0339735c51b43949809e972430#diff-ef9f64675d767bfa6e9c264105226575L2297) respects)